pkg/controller/logstorage/dashboards/dashboards_controller.go

```diff
@@ -0,0 +1,374 @@
+	"github.com/tigera/operator/pkg/controller/utils"
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/common/networkpolicy"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_logstorage_dashboards")
+
+const (
+	tigeraStatusName = "log-storage-dashboards"
+)
+
+type DashboardsSubController struct {
+	client          client.Client
+	scheme          *runtime.Scheme
+	status          status.StatusManager
+	provider        operatorv1.Provider
+	clusterDomain   string
+	usePSP          bool
+	multiTenant     bool
+	elasticExternal bool
+	tierWatchReady  *utils.ReadyFlag
+}
+
+func Add(mgr manager.Manager, opts options.AddOptions) error {
+	if !opts.EnterpriseCRDExists || opts.MultiTenant {
+		return nil
+	}
+
+	r := &DashboardsSubController{
+		client:         mgr.GetClient(),
+		scheme:         mgr.GetScheme(),
+		status:         status.New(mgr.GetClient(), tigeraStatusName, opts.KubernetesVersion),
+		clusterDomain:  opts.ClusterDomain,
+		provider:       opts.DetectedProvider,
+		tierWatchReady: &utils.ReadyFlag{},
+	}
+	r.status.Run(opts.ShutdownContext)
+
+	// Create a controller using the reconciler and register it with the manager to receive reconcile calls.
+	c, err := controller.New("log-storage-dashboards-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Determine how to handle watch events for cluster-scoped resources. For multi-tenant clusters,
+	// we should update all tenants whenever one changes. For single-tenant clusters, we can just queue the object.
+	var eventHandler handler.EventHandler = &handler.EnqueueRequestForObject{}
+	if opts.MultiTenant {
+		eventHandler = utils.EnqueueAllTenants(mgr.GetClient())
+	}
+
+	// Configure watches for operator.tigera.io APIs this controller cares about.
+	if err = c.Watch(&source.Kind{Type: &operatorv1.LogStorage{}}, eventHandler); err != nil {
+		return fmt.Errorf("log-storage-dashboards-controller failed to watch LogStorage resource: %w", err)
+	}
+	if err = c.Watch(&source.Kind{Type: &operatorv1.Installation{}}, eventHandler); err != nil {
+		return fmt.Errorf("log-storage-dashboards-controller failed to watch Installation resource: %w", err)
+	}
+	if err = c.Watch(&source.Kind{Type: &operatorv1.ManagementClusterConnection{}}, eventHandler); err != nil {
+		return fmt.Errorf("log-storage-dashboards-controller failed to watch ManagementClusterConnection resource: %w", err)
+	}
+	if err = utils.AddTigeraStatusWatch(c, "log-storage-dashboards"); err != nil {
```

Comment:
we should use `tigeraStatusName` name here

Reply:
Once @vara2504's PR is in, we'll need tdo add this status to the aggregated one as well. https://github.com/tigera/operator/pull/3118/